hatchery/tests/stacked.rs

```diff
@@ -56,3 +58,32 @@ pub fn multi_push_pop() -> Result<(), Error> {
 
     Ok(())
 }
+#[test]
+pub fn multi_push_store_restore_pop() -> Result<(), Error> {
+    let mut storage_path = PathBuf::new();
+    let first_id: ModuleId;
+    const N: i32 = 1_000;
+    {
+        let mut first_world = World::ephemeral()?;
+        first_id = first_world.deploy(module_bytecode!("stack"))?;
+        for i in 0..N {
+            first_world.transact(first_id, "push", i)?;
+            let len: i32 = first_world.query(first_id, "len", ())?;
+            assert_eq!(len, i + 1);
+        }
+        first_world.storage_path().clone_into(&mut storage_path);
+    }
+    let mut second_world = World::new(storage_path);
+    let second_id = second_world.deploy(module_bytecode!("stack"))?;
```

Comment:
Persisted worlds should include already deployed snapshots, and not have to re-deploy them.